src/bootstrap/build/step.rs

```diff
@@ -148,7 +148,9 @@ fn top_level(build: &Build) -> Vec<Step> {
             src: Source::Llvm { _dummy: () },
             target: &build.config.build,
         };
-        targets.push(t.doc(stage));
+        if build.config.docs {
+          targets.push(t.doc(stage));
```

Comment:
Done in latest commit. I'm assuming the commits will be squashed before being merged, right?